function/list_parameter_test.go

```diff
@@ -0,0 +1,251 @@
+		parameter function.ListParameter
+		expected  string
+	}{
+		"unset": {
+			parameter: function.ListParameter{},
+			expected:  "",
+		},
+		"Description-empty": {
+			parameter: function.ListParameter{
+				Description: "",
+			},
+			expected: "",
+		},
+		"Description-nonempty": {
+			parameter: function.ListParameter{
+				Description: "test description",
+			},
+			expected: "test description",
+		},
+	}
+
+	for name, testCase := range testCases {
+		name, testCase := name, testCase
+
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			got := testCase.parameter.GetDescription()
+
+			if diff := cmp.Diff(got, testCase.expected); diff != "" {
+				t.Errorf("unexpected difference: %s", diff)
+			}
+		})
+	}
+}
+
+func TestListParameterGetMarkdownDescription(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		parameter function.ListParameter
+		expected  string
+	}{
+		"unset": {
+			parameter: function.ListParameter{},
+			expected:  "",
+		},
+		"MarkdownDescription-empty": {
+			parameter: function.ListParameter{
+				MarkdownDescription: "",
+			},
+			expected: "",
+		},
+		"MarkdownDescription-nonempty": {
+			parameter: function.ListParameter{
+				MarkdownDescription: "test description",
+			},
+			expected: "test description",
+		},
+	}
+
+	for name, testCase := range testCases {
+		name, testCase := name, testCase
+
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			got := testCase.parameter.GetMarkdownDescription()
+
+			if diff := cmp.Diff(got, testCase.expected); diff != "" {
+				t.Errorf("unexpected difference: %s", diff)
+			}
+		})
+	}
+}
+
+func TestListParameterGetName(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		parameter function.ListParameter
+		expected  string
+	}{
+		"unset": {
+			parameter: function.ListParameter{},
+			expected:  function.DefaultParameterName,
+		},
+		"Name-empty": {
+			parameter: function.ListParameter{
+				Name: "",
+			},
+			expected: function.DefaultParameterName,
+		},
+		"Name-nonempty": {
+			parameter: function.ListParameter{
+				Name: "test",
+			},
+			expected: "test",
+		},
+	}
+
+	for name, testCase := range testCases {
+		name, testCase := name, testCase
+
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			got := testCase.parameter.GetName()
+
+			if diff := cmp.Diff(got, testCase.expected); diff != "" {
+				t.Errorf("unexpected difference: %s", diff)
+			}
+		})
+	}
+}
+
+func TestListParameterGetType(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		parameter function.ListParameter
+		expected  attr.Type
+	}{
+		"ElementType": {
+			parameter: function.ListParameter{
+				ElementType: basetypes.StringType{},
+			},
+			expected: basetypes.ListType{
+				ElemType: basetypes.StringType{},
+			},
+		},
+		// "CustomType": {
+		// 	parameter: function.ListParameter{
+		// 		CustomType: testtypes.ListType{},
+		// 	},
+		// 	expected: testtypes.ListType{},
+		// },
```

Comment:
Ah, its because the custom testing types did not implement the `Equal()` method. Implemented and uncommented these unit tests in https://github.com/hashicorp/terraform-plugin-framework/pull/889/commits/f1aa0f670ebd4adb7703b86a0561d0aa492de980 👍 Sorry about that.